gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ObservationCommentMapper.java

```diff
@@ -0,0 +1,148 @@
+                );
+
+                // Comment and Context may not always be mapped
+                setObservationComment(observation, narrativeStatement.getText());
+                setObservationContext(observation, ehrExtract, narrativeStatementId, encounters);
+
+                return observation;
+            }).toList();
+    }
+
+    private void setObservationContext(Observation observation, RCMRMT030101UK04EhrExtract ehrExtract,
+        II narrativeStatementId, List<Encounter> encounters) {
+        var composition =
+            EhrResourceExtractorUtil.extractEhrCompositionForNarrativeStatement(ehrExtract, narrativeStatementId);
+
+        encounters
+            .stream()
+            .filter(encounter -> encounter.getId().equals(composition.getId().getRoot()))
+            .findFirst()
+            .ifPresent(encounter -> observation.setContext(new Reference(encounter)));
+    }
+
+    private void setObservationComment(Observation observation, String text) {
+        if (!text.isBlank()) {
+            observation.setComment(text.trim());
+        }
+    }
+
+    private Meta createMeta() {
+        var meta = new Meta();
+        meta.setProfile(Collections.singletonList(new UriType(META_URL)));
+
+        return meta;
+    }
+
+    private Identifier createIdentifier(String narrativeStatementId) {
+        var identifier = new Identifier();
+        identifier.setSystem(IDENTIFIER_SYSTEM);
+        identifier.setValue(narrativeStatementId);
+
+        return identifier;
+    }
+
+    private InstantType createIssued(RCMRMT030101UK04EhrExtract ehrExtract, II narrativeStatementId) {
+        RCMRMT030101UK04EhrComposition composition =
+            EhrResourceExtractorUtil.extractEhrCompositionForNarrativeStatement(ehrExtract, narrativeStatementId);
+
+        if (composition.getAuthor().getTime().getNullFlavor() == null) {
+            return DateFormatUtil.parseToInstantType(composition.getAuthor().getTime().getValue());
+        }
+
+        return DateFormatUtil.parseToInstantType(ehrExtract.getAvailabilityTime().getValue());
+    }
+
+    private Reference createPerformer(RCMRMT030101UK04EhrExtract ehrExtract, RCMRMT030101UK04NarrativeStatement narrativeStatement) {
+        RCMRMT030101UK04EhrComposition composition =
+            EhrResourceExtractorUtil.extractEhrCompositionForNarrativeStatement(ehrExtract, narrativeStatement.getId());
+
+        return ParticipantReferenceUtil.getParticipantReference(narrativeStatement.getParticipant(), composition);
+    }
+
+    private CodeableConcept createCodeableConcept() {
+        var codeableConcept = new CodeableConcept();
+        codeableConcept.setCoding(
+            Collections.singletonList(new Coding(CODING_SYSTEM, CODING_CODE, CODING_DISPLAY)));
+
+        return codeableConcept;
+    }
+
+    private List<RCMRMT030101UK04EhrComposition> getCompositionsWithNarrativeStatement(RCMRMT030101UK04EhrExtract ehrExtract) {
```

Comment:
Could this method return narrative statements? Then you wouldn't have to do the filtering in the `mapObservations` method again

Reply:
Yup!